activestorage/test/jobs/transform_job_test.rb

```diff
@@ -16,6 +16,17 @@ class ActiveStorage::TransformJobTest < ActiveJob::TestCase
     end
   end
 
+  test "creates variant for previewable file" do
```

Comment:
hi @jonathanhefner @nicowenterodt 

thanks for your work.
 I would like to point out that this test does not test that a `variant` has been created. it tests that the preview has been generated

What I mean by that is that each `ActiveStorage::Blob` has a `has_one_attached :preview_image` which is another blob

and right now `processed` only generated this blob, **but not the variant that's associated with `preview_image`**

in my production app, I had to do this : 
```ruby
ActiveStorage::Preview.new(blob, transformations).processed.key
```

Adding the `key` call is a bit of hack, but it will not only generate the `preivew_image` but also the variant associated with which is, in my opinion, what's expected when we add the `preprocessed: true` ( the key that's returned is discarded, but it's the newly generated variant's key which i believe won't impact performance)

I will try to create a PR. I'm also considering changing the `processed` method implementation, but I feel it's a bit complex ( I need to further read on how to check if an image variant has already been processed for a set of transformation )

Reply:
I'm creating a PR

Reply:
Hi @chaadow, thanks for your feedback. 

So when I get it right this still does not really "preprocess" the actual variant you defined as a named variant like this:

```
has_one_attached :file do |attachable|
    attachable.preview :thumb, resize_to_limit: [400, 600], preprocessed: true
end
```

So there is no error now within the `TransformJob` because the preview_image will be preprocessed successfully utilizing the `representation` method.  **However** the defined variant of the preview itself is not respected and so when you access it like this:
 
 ```
 = image_tag my_record.file.representation(:thumb)
 ```
 
... the actual variant of the preview is missing which leads to a new attachment generated on the fly which is not really the purpose of being "preprocssed". 

Correct? I think I missed that because I focused too much on the `ActiveStorage::InvariableError` thrown within the `TransformJob` itself not checking the actual "preprocssed" characteristic of the defined variant. Oops. 

@jonathanhefner Is this worth a new issue? Or shall we reopen #50005 ?

Reply:
I've created a PR #50026

Reply:
Sorry i've added my comment without seeing yours

 @nicowenterodt  that's 100% correct. Since I had to manually pre process big HEIC files on rails 7.0 ( with a custom HEIC previewer, but logic is similar for any `previewable` blob), I realized that both the preview image and its associated variant were generated on the fly, and it created a bit of an unpleasant/suboptimal user experience.
 
I would love your feedback on my PR. thanks!

Reply:
Thanks @chaadow. I will have a look.